build_tools/waf_dynamo.py

```diff
@@ -469,33 +469,43 @@ def default_flags(self):
                 '-static-libstdc++'] + getAndroidLinkFlags(target_arch))
     elif 'web' == build_util.get_target_os():
 
-        emflags = ['DISABLE_EXCEPTION_CATCHING=1', 'AGGRESSIVE_VARIABLE_ELIMINATION=1', 'PRECISE_F32=2',
-                   'EXTRA_EXPORTED_RUNTIME_METHODS=["stringToUTF8","ccall","stackTrace","UTF8ToString","callMain"]',
-                   'ERROR_ON_UNDEFINED_SYMBOLS=1', 'INITIAL_MEMORY=33554432', 'LLD_REPORT_UNDEFINED', 'MAX_WEBGL_VERSION=2',
-                   'GL_SUPPORT_AUTOMATIC_ENABLE_EXTENSIONS=0']
+        emflags_compile = ['DISABLE_EXCEPTION_CATCHING=1']
+
+        emflags_compile = zip(['-s'] * len(emflags_compile), emflags_compile)
+        emflags_compile =[j for i in emflags_compile for j in i]
+
+        emflags_link = [
+            'DISABLE_EXCEPTION_CATCHING=1',
+            'EXPORTED_RUNTIME_METHODS=["ccall","stackTrace","UTF8ToString","callMain","HEAPU8"]',
+            'EXPORTED_FUNCTIONS=_malloc,_free',
+            'ERROR_ON_UNDEFINED_SYMBOLS=1',
+            'INITIAL_MEMORY=33554432',
+            'MAX_WEBGL_VERSION=2',
+            'GL_SUPPORT_AUTOMATIC_ENABLE_EXTENSIONS=0',
+            'STACK_SIZE=5MB']
```

Comment:
@ekharkunov You directly stated the stack size. What's the default value in Emscripten?

Reply:
Here marked that default STACK_SIZE was changed from 5Mb to 64Kb https://github.com/emscripten-core/emscripten/blob/3.1.55/ChangeLog.md#3127---112922
```
The default STACK_SIZE was reduced from 5MB to 64KB. Projects that use more than 64Kb of stack will now need specify -sSTACK_SIZE at link time. For example, -sSTACK_SIZE=5MB can be used to restore the previous behaviour. To aid in debugging, as of #18154, we now also place the stack first in memory in debug builds so that overflows will be immediately detected, and result in runtime errors. This change brings emscripten into line with wasm-ld and wasi-sdk defaults, and also reduces memory usage by default. In general, WebAssembly stack usage should be lower than on other platforms since a lot of state normally stored on the stack is hidden within the runtime and does not occupy linear memory at all. The default for DEFAULT_PTHREAD_STACK_SIZE was also reduced from 2MB to 64KB to match.
```
That's why I specified explicitly STACK_SIZE. Later I'll add task to make it configurable via project settings.